vocode/streaming/synthesizer/eleven_labs_synthesizer.py

```diff
@@ -48,6 +77,117 @@ def __init__(
         self.optimize_streaming_latency = synthesizer_config.optimize_streaming_latency
         self.words_per_minute = 150
         self.experimental_streaming = synthesizer_config.experimental_streaming
+        self.uri = f"wss://{ELEVENLABS_BASE_URL}/text-to-speech/{self.voice_id}/stream-input?model_id={self.model_id}"
+
+    async def create_text_generator(self):
+        # yield "Hi. How are you doing."
+        # yield "What can I do for you"
+        first_receive = False
+        while True:
+            try:
+                text = await self.text_queue.get()
+                if not first_receive:
+                    self.logger.debug("Got first stream text")
+                    first_receive = True
+                if text is None:
+                    self.logger.debug("Got end stream text")
+                    break
+                self.logger.debug(f"Stream text: {text}")
+                yield text
+            except asyncio.CancelledError:
+                self.logger.warn("Canceled: create_text_generator")
+                break
+        self.text_queue = None
+        self.text_generator = None
+
+    async def generate_speech_output_chunks(self):
+        async with websockets.connect(self.uri) as ws:
+            await ws.send(
+                json.dumps(
+                    {
+                        "text": " ",
+                        "voice_settings": {
+                            "stability": self.stability,
+                            "similarity_boost": self.similarity_boost,
+                        },
+                        "xi_api_key": self.api_key,
+                    }
+                )
+            )
+            async for text in text_chunker(self.text_generator):
+                self.logger.debug(f"sending text over socket: {text}")
+                await ws.send(
+                    json.dumps({"text": text, "try_trigger_generation": True})
+                )
+            await ws.send(json.dumps({"text": ""}))
+            """Listen to the websocket for audio data and stream it."""
+            while True:
+                try:
+                    msg = await ws.recv()
+                    data = json.loads(msg)
+                    if data.get("audio"):
+                        yield base64.b64decode(data["audio"])
+                    elif data.get("isFinal"):
+                        break
+                except websockets.ConnectionClosed:
+                    print("Connection closed")
+                    break
+            self.logger.debug(f"socket may get closed now")
+
+    async def create_speech_stream(self, text: str, is_end: bool = False):
```

Comment:
How is this function being used? Shouldn't it be called in streaming_conversation?

Reply:
Thanks for calling out. I had missed to add those changes. Added them now. Please see streaming_conversation.py